trustmanager/filestore.go

```diff
@@ -61,7 +69,52 @@ func (f *fileStore) Add(name string, data []byte) error {
 	return ioutil.WriteFile(filePath, data, f.perms)
 }
 
-// Remove removes a file identified by name
+// AddEncrypted writes encrypted data to a file with a given name, given a key
+func (f *fileStore) AddEncrypted(name string, data []byte, passphrase string) error {
+	filePath := f.genEncryptedFilePath(name)
+	createDirectory(filepath.Dir(filePath), f.perms)
+
+	// Derive a key from the passphrase
+	// First generate random salt for scrypt
+	salt := make([]byte, SaltSize)
+	_, err := rand.Read(salt)
+	if err != nil {
+		return err
+	}
+
+	// With the salt, generate key derived from passphrase
+	derivedKey, err := scrypt.Key([]byte(passphrase), salt, 16384, 8, 1, KeySize)
```

Comment:
Should those integer constants be declared at the top of the file for clarity? They're reused below.